craft_providers/bases/buildd.py

```diff
@@ -100,6 +122,33 @@ def __init__(
 
         self.hostname = hostname
 
+    def _ensure_instance_config_compatible(
+        self, *, executor: Executor, deadline: Optional[float]
+    ) -> None:
+        """Ensure instance configuration is compatible.
+
+        :raises BaseCompatibilityError: if instance is incompatible.
+        :raises BaseConfigurationError: on other unexpected error.
+        """
+        _check_deadline(deadline)
+        config = self.instance_config_type.load(
+            executor=executor,
+            config_path=self.instance_config_path,
+        )
+
+        # If no config has been written, assume it is compatible (likely an
+        # unfinished setup).
+        if config is None:
+            return
```

Comment:
Is this safe? What are the consequences of assuming it's compatible when it's not?

Reply:
In the case of the instance config, if it's not present, we're dependent on the other checks that are done (i.e. os-release check).